src/gentropy/susie_finemapper.py

```diff
@@ -0,0 +1,140 @@
+            sorted_arr = susie_result[
+                susie_result[:, i + 1].astype(float).argsort()[::-1]
+            ]
+            cumsum_arr = np.cumsum(sorted_arr[:, i + 1].astype(float))
+            filter_row = np.argmax(cumsum_arr >= 0.99)
+            if filter_row == 0 and cumsum_arr[0] < 0.99:
+                filter_row = len(cumsum_arr)
+            filter_row += 1
+            filtered_arr = sorted_arr[:filter_row]
+            cred_set = filtered_arr[:, [0, i + 1, i + L_snps + 1, i + 2 * L_snps + 1]]
+            win = Window.rowsBetween(
+                Window.unboundedPreceding, Window.unboundedFollowing
+            )
+            cred_set = (
+                session.spark.createDataFrame(
+                    cred_set.tolist(),
+                    ["variantId", "posteriorProbability", "logBF", "beta"],
+                )
+                .join(
+                    _join.select(
+                        "variantId",
+                        "chromosome",
+                        "position",
+                    ),
+                    "variantId",
+                )
+                .sort(f.desc("posteriorProbability"))
+                .withColumn(
+                    "locus",
+                    f.collect_list(
+                        f.struct(
+                            f.col("variantId").cast("string").alias("variantId"),
+                            f.col("posteriorProbability")
+                            .cast("double")
+                            .alias("posteriorProbability"),
+                            f.col("logBF").cast("double").alias("logBF"),
+                            f.col("beta").cast("double").alias("beta"),
+                        )
+                    ).over(win),
+                )
+                .limit(1)
+                .withColumns(
+                    {
+                        "studyId": f.lit(_studyId),
+                        "region": f.lit(_region),
+                        "credibleSetIndex": f.lit(counter),
+                        "credibleSetlog10BF": f.lit(value * 0.4342944819),
+                        "finemappingMethod": f.lit("SuSiE-inf"),
+                    }
+                )
+                .withColumn(
+                    "studyLocusId",
+                    StudyLocus.assign_study_locus_id(
+                        f.col("studyId"), f.col("variantId")
+                    ),
+                )
+                .select(
+                    "studyLocusId",
+                    "studyId",
+                    "region",
+                    "credibleSetIndex",
+                    "locus",
+                    "variantId",
+                    "chromosome",
+                    "position",
+                    "finemappingMethod",
+                    "credibleSetlog10BF",
+                )
+            )
+            if cred_sets is None:
+                cred_sets = cred_set
+            else:
+                cred_sets = cred_sets.union(cred_set)
```

Comment:
I would use `unionByName` rather `union`, because union concatenates columns by positions instead of names. I think in this case it is fine, as the order of columns are defined in rwo 120, but still.